src/hooks/useUndernames/useUndernames.tsx

```diff
@@ -23,10 +23,13 @@ export function useUndernames(id?: ArweaveTransactionID) {
   const arweaveDataProvider = useArweaveCompositeProvider();
   const [sortAscending, setSortOrder] = useState(true);
   const [sortField, setSortField] = useState<keyof UndernameMetadata>('status');
-  const [selectedRow, setSelectedRow] = useState<UndernameMetadata>(); // eslint-disable-line
+  const [selectedRow, setSelectedRow] = useState<UndernameMetadata>();
   const [rows, setRows] = useState<UndernameMetadata[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const [percent, setPercentLoaded] = useState<number>(0);
+  const [action, setAction] = useState<
+    'create' | 'edit' | 'remove' | undefined
```

Comment:
move these to a shared type array `const UndernameInteractions = ['create', 'remove', 'edit'] as const`

Reply:
added